internal/app/machined/pkg/system/services/apid.go

```diff
@@ -164,3 +179,32 @@ func (o *APID) HealthFunc(runtime.Runtime) health.Check {
 func (o *APID) HealthSettings(runtime.Runtime) *health.Settings {
 	return &health.DefaultSettings
 }
+
+func (o *APID) syncKubeletPKI() {
+	if err := copy.Dir(constants.KubeletPKIDir, constants.SystemPKIDir); err != nil {
```

Comment:
We might want to create explicitly a directory first say with 0700 permissions to at least partially restrict access to sensitive data